### ISSUE TITLE
bug: `requestAnimationFrame` zone grows forever

### ISSUE THREAD

**Issue by justindujardin** [angular/zone.js#143]
When using a basic requestAnimationFrame render loop, the zone grows forever. 
Related to https://github.com/angular/angular/issues/2912

```javascript
zone.fork().run(function () {
    function frameCallback() {
        // frame++
        requestAnimationFrame(frameCallback);
    }
    requestAnimationFrame(frameCallback);
});
```

Example: https://jsfiddle.net/9dqzky6g/3/

**Comment by justindujardin** [comment]
It looks like subsequent calls to `requestAnimationFrame` create child Zones, so unless your `requestAnimationFrame` loop stops, you will add a child zone for each invocation.

zoneBoundFn eats up all of the profile idle time in a few thousand frames.

**Comment by justindujardin** [comment]
Anyone care to comment on this?  It is making it hard for me to run my app for more than a few minutes with Angular2.  Everything is wonderfully fast, until the zones start piling up.

I'm willing to look at fixing the issue if I could get some direction on where to go with it.

**Comment by justindujardin** [comment]
Updated example with patch applied: https://jsfiddle.net/cg9fa5nf/1/

**Comment by vicb** [comment]
closed via #147

**Comment by btford** [comment]
:clap: Thanks @vicb
